gameplay/avalonRoom.js

```diff
@@ -1388,11 +1389,24 @@ module.exports = function (host_, roomId_, io_, maxNumPlayers_, newRoomPassword_
 			}
 		}
 
-
 		// If they have not been kicked before
-		if (thisRoom.kickedPlayers[socket.request.user.username] === true) {
-			socket.emit("danger-alert", "You have been banned from this room. You can not join.");
-			return false;
+		if (Object.keys(thisRoom.kickedPlayers).includes(socket.request.user.username) == true){
+			if(thisRoom.kickedPlayers[socket.request.user.username]["status"] === true) {
+				var banTimeLeft = banTime - ((new Date).getTime() - thisRoom.kickedPlayers[socket.request.user.username]["startTime"]);
+				if (banTimeLeft > 0) {
+					var banMsg = "You have been banned from joining the game for the next " + 
+						convertMillisToSeconds(banTimeLeft) + ". You can not join yet.";
+					socket.emit("danger-alert", banMsg);
+					return false;
+				} else {
+					//if ban time has been depleted, allow player to join the room again
+					//by removing him from the kickedPlayers map
+					delete thisRoom.kickedPlayers[socket.request.user.username];
```

Comment:
Please use Array.prototype.splice instead of delete.

Reply:
Accepted. Will be addressed in the next commit.